dashboard/app/lib/stylesheets/dashboard/apps.scss

```diff
@@ -12,3 +12,6 @@
   overflow-y: auto;
 }
 
+.system-applications-link {
+  float: right;
```

Comment:
Wrap in a `<div>` and set `text-align: right` on it (a clear fix is needed otherwise).